lib/core/src/restore_onchain.rs

```diff
@@ -0,0 +1,334 @@
+/// Methods to simulate the immutable DB data available from real-time sync
+// TODO Remove once real-time sync is integrated
+pub(crate) mod immutable {
+    use std::collections::HashMap;
+
+    use anyhow::Result;
+    use boltz_client::boltz::CreateReverseResponse;
+    use boltz_client::LBtcSwapScript;
+    use log::{error, info};
+    use lwk_wollet::elements::Script;
+
+    use crate::sdk::LiquidSdk;
+
+    pub(crate) struct ImmutableDb {
```

Comment:
NIT: Wouldn't use immuntable as it doesn't hint anytning about the data it contains. Perhaps just SwapsList ?